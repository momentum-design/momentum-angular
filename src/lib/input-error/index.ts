export * from './input-error.module';
export * from './input-error.component';
