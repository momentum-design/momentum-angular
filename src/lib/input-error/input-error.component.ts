import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'md-input-error',
  template: `
    <div class="md-input__messages" role="alert">
      <div class="message">{{ error }}</div>
    </div>
  `,
  styles: [],
})
export class InputErrorComponent implements OnInit {
  @Input() public error: string;

  constructor() {}

  ngOnInit() {}
}
