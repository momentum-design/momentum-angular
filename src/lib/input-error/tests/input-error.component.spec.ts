import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { InputErrorComponent } from '../input-error.component';

describe('InputErrorComponent', () => {
  let component: InputErrorComponent;
  let fixture: ComponentFixture<InputErrorComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [InputErrorComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InputErrorComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render a error message with correct text', () => {
    component.error = 'This is where the success message would be.';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
    const inputNativeElement = fixture.nativeElement;
    const error = inputNativeElement.querySelector('.message');

    expect(error.textContent).toBe('This is where the success message would be.');
  });
});
