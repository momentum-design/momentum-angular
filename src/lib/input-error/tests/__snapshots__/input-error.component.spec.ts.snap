// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputErrorComponent should create and match snapshot 1`] = `
<md-input-error>
  <div
    class="md-input__messages"
    role="alert"
  >
    <div
      class="message"
    >

    </div>
  </div>
</md-input-error>
`;

exports[`InputErrorComponent should render a error message with correct text 1`] = `
<md-input-error
  error={[Function String]}
>
  <div
    class="md-input__messages"
    role="alert"
  >
    <div
      class="message"
    >
      This is where the success message would be.
    </div>
  </div>
</md-input-error>
`;
